.github/DEVELOPMENT.md

```diff
@@ -0,0 +1,134 @@
+# Development
+
+Developers should read [the development section of the website](https://trino.io/development),
+which covers things like development philosophy and contribution process.
+
+* [Getting started](#getting-started)
+* [Running tests](#running-tests)
+* [Commits and pull requests](#commits-and-pull-requests)
+* [Code style](#code-style)
+* [Releasing](#releasing)
+
+## Getting started
+
+Start by forking the repository and then modify the code in your fork.
+
+We recommend that you use Python3's `venv` for development:
+
+```
+$ python3 -m venv .venv
+$ . .venv/bin/activate
+$ pip install -e '.[tests]'
+```
+
+With `-e` passed to `pip install` above pip can reference the code you are
+modifying in the *virtual env*. That way, you do not need to run `pip install`
+again to make your changes applied to the *virtual env*.
+
+## Running tests
+
+`trino-python-client` uses [pytest](https://pytest.org/) for its tests. To run
+only unit tests, type:
+
+```
+$ pytest tests/unit
+```
+
+Then you can pass options like `--pdb` or anything supported by `pytest --help`.
+
+To run integration tests:
+
+```
+$ pytest tests/integration
+```
+
+They pull a Docker image and then run a container with a Trino server:
+- the image is named `trinodb/trino:${TRINO_VERSION}`
+- the container is named `trino-python-client-tests-{uuid4()[:7]}`
+
+To run the tests with different versions of Python in managed *virtual envs*,
+use `tox` (see the configuration in `tox.ini`):
+
+```
+$ tox
+```
+
+When the code is ready, submit a Pull Request.
```

Comment:
Perhaps indicating somewhere that creating an issue before actually starting working on something make sense to avoid unnecessary work which won't be accepted by maintainers?

Reply:
One of the first things linked from this DEVELOPMENT.md page - https://trino.io/development/process.html